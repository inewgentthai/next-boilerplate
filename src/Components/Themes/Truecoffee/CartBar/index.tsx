// #region Global Imports
import * as React from "react";
// #endregion Global Imports

// #region Local Imports
import { ICartBar } from "./CartBar";
// #endregion Local Imports

const CartBar: React.FunctionComponent<ICartBar.IProps> = (): JSX.Element => {
    return (
      <section className="cart-bar fixed-bottom">
        <div className="container">
          <a href="cart.html">
            <div className="row no-gutters justify-content-between align-items-center">
              <div className="col-4"><div className="item-in-cart badge">3</div></div>
              <div className="col go-to-cart">Go to Cart</div>
              <div className="col-4 total">
                <div>Total</div>
                <div>à¸¿9,999.00</div>
              </div>
            </div>
          </a>
        </div>
      </section>
    );
};

export { CartBar };
