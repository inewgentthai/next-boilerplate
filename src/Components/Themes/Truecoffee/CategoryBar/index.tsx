// #region Global Imports
import * as React from "react";
// #endregion Global Imports

// #region Local Imports
import { ICategoryBar } from "./CategoryBar";
// #endregion Local Imports

const CategoryBar: React.FunctionComponent<ICategoryBar.IProps> = (): JSX.Element => {
    return (
      <section className="categories">
        <div className="container">
          <div className="categories-list row justify-content-center">
            <div className="item capsule"><a href="#">Allsss</a></div>
            <div className="item capsule active"><a href="#">Coffee</a></div>
            <div className="item capsule"><a href="#">Tea</a></div>
            <div className="item capsule"><a href="#">Non Coffee</a></div>
            <div className="item capsule"><a href="#">Tea</a></div>
            <div className="item capsule"><a href="#">Non Coffee</a></div>
            <div className="item capsule"><a href="#">Tea</a></div>
          </div>
          <div className="row">
            <div className="col cat-found text-center"><b className="text-red">“Coffee”</b> Found 8 item</div>
          </div>
        </div>
      </section>
    );
};

export { CategoryBar };
