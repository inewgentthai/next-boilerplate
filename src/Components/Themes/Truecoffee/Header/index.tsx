// #region Global Imports
import * as React from "react";
import Head from 'next/head';
// #endregion Global Imports

// #region Local Imports
import "./style.scss";
// #endregion Local Imports

// #region Interface Imports
import { IHeader } from "./Header";
// #endregion Interface Imports

const Header: React.FunctionComponent<IHeader.IProps> = (
    props
): JSX.Element => {
    const { text } = props;

    return (
        <Head>
          <title>{ text } </title>
          <meta property="og:title" content={ text } key="title" />
        </Head>
    );
};

export { Header };
