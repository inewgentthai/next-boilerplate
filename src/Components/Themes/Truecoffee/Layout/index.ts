export { DefaultLayout } from "./Default";
export { HomeLayout } from "./Home";