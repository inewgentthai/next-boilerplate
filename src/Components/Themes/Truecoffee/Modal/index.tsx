// #region Global Imports
import * as React from "react";
// #endregion Global Imports

// #region Local Imports
import { IModal } from "./Modal";
// #endregion Local Imports

const Modal: React.FunctionComponent<IModal.IProps> = (): JSX.Element => {
    return (
      <div className="modal fade show" id="itemModal" tabIndex={-1} role="dialog" aria-labelledby="itemModal" aria-hidden="false">
        <div className="item-modal modal-dialog" role="document">
          <div className="modal-content">
            <div className="paper">
              <div className="product-thumb-box">
                <div className="icn-close" data-dismiss="modal" aria-label="Close" />
                <img src="static/truecoffee-ordering/images/product-coffee1.png" srcSet="static/truecoffee-ordering/images/product-coffee1.png 2x, static/truecoffee-ordering/images/product-coffee1.png 3x" className="product-thumb" />
              </div>
              <div className="product-info">
                <div className="product-type">Coffee</div>
                <div className="product-name">Iced Cappuccino - M</div>
                <div className="product-desc">Our dark, rich espresso is combined with milk and served over ice. A perfect milk forward cool down.</div>
                <div className="product-price">฿85</div>
                <div className="product-quantity d-flex justify-content-center">
                  <a className="icn-minus disable" />
                  <input type="tel" className="form-control" id="itemquantity" min={1} max={99} maxLength={2} />
                  <a className="icn-plus" />
                </div>
                <div className="product-note d-flex justify-content-center">
                  <div className="form-group text-center">
                    <label htmlFor="itemnote">Additional Information</label>
                    <input type="text" className="form-control" id="itemnote" placeholder="เช่น ระดับความหวาน,น้ำแข็ง ฯลฯ" />
                  </div>
                </div>
              </div>
              <div className="btn-set">
                <button className="btn btn-red btn-block" >Add item to Cart</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
};

export { Modal };
