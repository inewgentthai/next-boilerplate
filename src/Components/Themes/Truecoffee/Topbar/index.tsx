// #region Global Imports
import * as React from "react";
// #endregion Global Imports

// #region Local Imports
import { ITopbar } from "./Topbar";
// #endregion Local Imports

const Topbar: React.FunctionComponent<ITopbar.IProps> = (): JSX.Element => {
    return (
      <section className="topbar fixed-top">
        <div className="container">
          <div className="row no-gutters">
            <div className="col text-left">
              <div className="icn-history float-left">
                <a href="order-history.html" data-toggle="tooltip" data-placement="right" title="Order History" /></div>
            </div>
            <div className="col-6 text-center">
              <a href="index.html"><img src="static/truecoffee-ordering/images/logo.svg" /></a>
            </div>
            <div className="col text-right">
              <div className="icn-cart float-right">
                <a href="cart.html"><div className="icn-noti">1</div></a>
              </div>
            </div>
          </div>
        </div>
      </section>
    );
};

export { Topbar };
