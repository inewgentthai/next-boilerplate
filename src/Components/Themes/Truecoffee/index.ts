export { Layout } from "./Layout";
export { Topbar } from "./Topbar";
export { CategoryBar } from "./CategoryBar";
export { CartBar } from "./CartBar";
export { Modal } from "./Modal";
export { Footer } from "./Footer";
export { Header } from "./Header";
