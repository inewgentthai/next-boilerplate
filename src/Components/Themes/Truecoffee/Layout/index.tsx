// #region Global Imports
import * as React from "react";
// #endregion Global Imports

// #region Local Imports
import { ILayout } from "./Layout";
import { CartBar } from "../CartBar";
import { CategoryBar } from "../CategoryBar";
import { Topbar } from "../Topbar";
import { Modal } from "../Modal";
import { Header } from "../Header";
import { Footer } from "@Components/Footer";
// #endregion Local Imports

const Layout: React.FunctionComponent<ILayout.IProps> = ({ children, title = 'True Coffee : Smart Ordering' }): JSX.Element => {
    return (
     <div>
        <Header text={ title }/>
        <div className="loading-icon" />
        <div className="wrapper">
          {/* Topbar */}
          <Topbar />
          {/* End Topbar */}
          {/* Categories */}
          <CategoryBar />
          {/* End Categories */}
          {/* Item List */}
          { children }
          {/* End Item List */}
          {/* Cart Bar */}
          <CartBar />
          {/* End Cart Bar */}
          {/* Modal */}
          <Modal />
          {/* End Modal */}
        </div>
        <Footer />
      </div>
    );
};

export { Layout };
