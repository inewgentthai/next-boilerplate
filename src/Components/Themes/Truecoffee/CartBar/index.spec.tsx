// #region Global Imports
import * as React from "react";
// #endregion Global Imports

// #region Local Imports
import { render } from "@Test/utils";
import { CartBar } from "@Components/Themes/Truecoffee";
// #endregion Local Imports

describe("CartBar", () => {
    it("should match snapshot", () => {
        const { container } = render(<CartBar>Test</CartBar>);

        expect(container).toMatchSnapshot();
    });
});
