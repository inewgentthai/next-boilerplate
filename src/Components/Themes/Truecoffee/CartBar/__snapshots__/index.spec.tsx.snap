// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartBar should match snapshot 1`] = `
<div>
  <section
    class="cart-bar fixed-bottom"
  >
    <div
      class="container"
    >
      <a
        href="cart.html"
      >
        <div
          class="row no-gutters justify-content-between align-items-center"
        >
          <div
            class="col-4"
          >
            <div
              class="item-in-cart badge"
            >
              3
            </div>
          </div>
          <div
            class="col go-to-cart"
          >
            Go to Cart
          </div>
          <div
            class="col-4 total"
          >
            <div>
              Total
            </div>
            <div>
              à¸¿9,999.00
            </div>
          </div>
        </div>
      </a>
    </div>
  </section>
</div>
`;
